docs/scripts/build_examples.py

```diff
@@ -0,0 +1,218 @@
+    # examples
+    ex_file = """# Examples\n\n```{toctree}\n:maxdepth: 2\n\n"""
+
+    # List all directories in the examples folder
+    ex_directories = [d for d in destination_directory.iterdir() if d.is_dir()]
+
+    # This list contains the order of the examples as we want to have them in the end.
+    # The examples that should be the first ones are already included here and skipped
+    # later on. ALl other are just included.
+    ex_order = [
+        "Basics<Basics/Basics>\n",
+        "Searchspaces<Searchspaces/Searchspaces>\n",
+        "Constraints Discrete<Constraints_Discrete/Constraints_Discrete>\n",
+        "Constraints Continuous<Constraints_Continuous/Constraints_Continuous>\n",
+        "Multi Target<Multi_Target/Multi_Target>\n",
+        "Serialization<Serialization/Serialization>\n",
+        "Custom Surrogates<Custom_Surrogates/Custom_Surrogates>\n",
+    ]
+
+    # Iterate over the directories.
+    for sub_directory in (pbar := tqdm(ex_directories)):
+        # Get the name of the current folder
+        # Format it by replacing underscores and capitalizing the words
+        folder_name = sub_directory.stem
+        formatted = " ".join(word.capitalize() for word in folder_name.split("_"))
+
+        # Create the link to the folder to the top level toctree.
+        ex_file_entry = formatted + f"<{folder_name}/{folder_name}>\n"
+        # Add it to the list of examples if it is not already contained
+        if ex_file_entry not in ex_order:
+            ex_order.append(ex_file_entry)
+
+        # We need to create a file for the inclusion of the folder.
+        # We thus get the content of the corresponding header file.
+        header_folder_name = sub_directory / f"{folder_name}_Header.md"
+        header = header_folder_name.read_text()
+
+        subdir_toctree = header + "\n```{toctree}\n:maxdepth: 1\n\n"
+
+        # Set description of progressbar
+        pbar.set_description("Overall progress")
+
+        # list all .py files in the subdirectory that need to be converted
+        py_files = list(sub_directory.glob("**/*.py"))
+
+        # Iterate through the individual example files
+        for file in (inner_pbar := tqdm(py_files, leave=False)):
+            # Include the name of the file to the toctree
+            # Format it by replacing underscores and capitalizing the words
+            file_name = file.stem
+
+            formatted = " ".join(word.capitalize() for word in file_name.split("_"))
+            # Remove duplicate "constraints" for the files in the constraints folder.
+            if "Constraints" in folder_name and "Constraints" in formatted:
+                formatted = formatted.replace("Constraints", "")
+
+            # Also format the Prodsum name to Product/Sum
+            if "Prodsum" in formatted:
+                formatted = formatted.replace("Prodsum", "Product/Sum")
+            subdir_toctree += formatted + f"<{file_name}>\n"
```

Comment:
I don't like the fact very much that we have these hard-coded rules sitting somewhere in the middle of the rather lengthy processing logic. I'm not asking you to change it here – rather I think this is something that can  be fixed during an example overhaul. Just pointing at the problem already...